VideoApp/VideoApp/NewCode/Screens/Room/RoomViewModel.swift

```diff
@@ -81,4 +84,42 @@ import Combine
         roomManager.localParticipant.isMicOn = false
         roomManager.localParticipant.isCameraOn = false
     }
+
+    /// A user can manually switch layout to override auto layout.
+    func switchToLayout(_ layout: Layout) {
+        self.layout = layout
+
+        switch layout {
+        case .grid:
+            break
+        case .focus:
+            /// Turn auto layout back on when there is a presentation and the user switched back to focus layout
+            isAutoLayoutSwitchingEnabled = focusLayoutViewModel.isPresenting
+        }
+    }
+    
+    private func handleIsPresentingChange(isPresenting: Bool) {
+        if isPresenting {
+            switch layout {
+            case .grid:
+                /// The grid does not show the presentation at all so we have to auto switch to
+                /// focus to inform the user that a presentation has started
+                isAutoLayoutSwitchingEnabled = true
+            case .focus:
+                break
+            }
+
+            layout = .focus
+        } else {
+            if isAutoLayoutSwitchingEnabled {
+                layout = .grid
+            }
+        }
+    }
```

Comment:
There is some interesting logic here to switch between manual and auto layout switching modes. So good to review this and make sure it makes sense.